telegram/ext/_conversationhandler.py

```diff
@@ -170,54 +209,54 @@ class ConversationHandler(Handler[Update, CCT]):
                  from what you expect.
 
 
-        name (:obj:`str`, optional): The name for this conversationhandler. Required for
+        name (:obj:`str`, optional): The name for this conversation handler. Required for
             persistence.
         persistent (:obj:`bool`, optional): If the conversations dict for this handler should be
             saved. Name is required and persistence has to be set in :class:`telegram.ext.Updater`
         map_to_parent (Dict[:obj:`object`, :obj:`object`], optional): A :obj:`dict` that can be
-            used to instruct a nested conversationhandler to transition into a mapped state on
-            its parent conversationhandler in place of a specified nested state.
-        run_async (:obj:`bool`, optional): Pass :obj:`True` to *override* the
-            :attr:`Handler.run_async` setting of all handlers (in :attr:`entry_points`,
+            used to instruct a nested conversation handler to transition into a mapped state on
+            its parent conversation handler in place of a specified nested state.
+        block (:obj:`bool`, optional): Pass :obj:`False` to *overrule* the
+            :attr:`Handler.block` setting of all handlers (in :attr:`entry_points`,
             :attr:`states` and :attr:`fallbacks`).
-
-            Note:
-                If set to :obj:`True`, you should not pass a handler instance, that needs to be
-                run synchronously in another context.
+            Defaults to :obj:`True`, in which case the handlers setting will be respected.
 
             .. versionadded:: 13.2
+            .. versionchanged:: 14.0
+                No longer *overrides* the handlers settings.
```

Comment:
Uff, I recall rewriting this a few times … before getting into doc & implementation details, I guess we should clarify what the desired behavior is. I guess we should always use CHs internal non-blocking behavior if `CH.handle_update` itself is called in a non-blocking way by `App.process_update` - to make sure that the next incoming update can be handled correctly. IISC this then gives us this table:

<details>

| **Defaults.block** | **CH.block** | **handler.block** | **Does App.process_update use create_task on CH?** | **Use create_task in CH?** | **comments**                          |
|--------------------|--------------|-------------------|----------------------------------------------------|----------------------------|---------------------------------------|
| True               | Default True | Default True      | ✘                                                  | ✘                          |                                       |
| True               | Default True | True              | ✘                                                  | ✘                          |                                       |
| True               | Default True | False             | ✘                                                  | ✔                          |                                       |
| True               | True         | Default True      | ✘                                                  | ✘                          |                                       |
| True               | True         | True              | ✘                                                  | ✘                          |                                       |
| True               | True         | False             | ✘                                                  | ✘                          |  Here CH overrides Handler            |
| True               | False        | Default True      | ✔                                                  | ✔                          |  Here CH overrides Handler            |
| True               | False        | True              | ✔                                                  | ✔                          |  Here CH overrides Handler            |
| True               | False        | False             | ✔                                                  | ✔                          |                                       |
| False              | Default True | Default True      | ✔                                                  | ✔                          | Here App overrides Handler indirectly |
| False              | Default True | True              | ✔                                                  | ✔                          | Here App overrides Handler indirectly |
| False              | Default True | False             | ✔                                                  | ✔                          |                                       |
| False              | True         | Default True      | ✘                                                  | ✘                          |                                       |
| False              | True         | True              | ✘                                                  | ✘                          |                                       |
| False              | True         | False             | ✘                                                  | ✘                          |  Here CH overrides Handler            |
| False              | False        | Default True      | ✔                                                  | ✔                          |  Here CH overrides Handler            |
| False              | False        | True              | ✔                                                  | ✔                          |  Here CH overrides Handler            |
| False              | False        | False             | ✔                                                  | ✔                          |


</details>

Does that make sense to you?

Reply:
Yes, table makes 100% sense to me. 

Maybe we should directly include the table in the docs/wiki?

Reply:
actually, on second thought, it might be better to have `App` just awlays call `CH.handle_update` in a blocking way, as `CH` takes care of the non-blocking logic internally. we could do that along the lines

```python
__init__(self, block= DF_TRUE):
    self.block = True  # Such that App.process_update will never try to run in non-blocking
    self._block = block

def handle_update(self, …)
    if self._block is not DF_TRUE:
        block = self._block
    else:
        if handler.block is DF_TRUE:
            block = application.bot.defaults.block
        else:
            block = handler.block
```

such that the logic is:

* if you pass `block` to CH, it overrules the handlers settings
* otherwise the handlers behave just like in `process_update`, including defaults-logic